
ackermann_steering_controller:
  type: "ackermann_steering_controller/AckermannSteeringController"
  rear_wheel  : 'rear_wheel_joint'
  front_steer : 'front_steer_joint'
  publish_rate: 100.0 # defaults to 50

  pose_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
  twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

  #wheel_separation_h : 0.4
  #wheel_radius : 0.11
  wheel_separation_h_multiplier: 1.0 # calibration parameter for odometory, needed for test.

  # Velocity commands timeout [s], default 0.5
  cmd_vel_timeout: 0.25

  # Base frame_id
  base_frame_id: atlascar2/base_footprint #default: base_link

  # Odom frame_id
  odom_frame_id: atlascar2/odom

  # Velocity and acceleration limits
  # Whenever a min_* is unspecified, default to -max_*
  linear:
    x:
      has_velocity_limits: true
      max_velocity: 1.0  # m/s
      min_velocity: -1.0 # m/s
      has_acceleration_limits: true
      max_acceleration: 0.8  # m/s^2
      min_acceleration: -0.4 # m/s^2
      has_jerk_limits: true
      max_jerk: 5.0 # m/s^3

  angular:
    z:
      has_velocity_limits: true
      max_velocity: 1.5  # rad/s
      min_velocity: -1.5  # rad/s
      has_acceleration_limits: true
      max_acceleration: 1.5  # rad/s^2
      has_jerk_limits: true
      max_jerk: 2.5 # rad/s^3